core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/ast/PropertyShape.java

```diff
@@ -282,44 +283,36 @@ public SparqlFragment buildSparqlValidNodes_rsx_targetShape(StatementMatcher.Var
 		StatementMatcher.Variable someObject = new StatementMatcher.Variable(
 				UUID.randomUUID().toString().replace("-", ""));
 
-		boolean isFilterCondition = constraintComponents.stream()
-				.map(o -> o.buildSparqlValidNodes_rsx_targetShape(object, someObject, rdfsSubClassOfReasoner,
+		List<SparqlFragment> sparqlFragments = constraintComponents.stream()
```

Comment:
Is it worth pulling those changes into a single class to reduce code duplication?

Reply:
Since I cleaned up this code quite a bit it has gotten very short and concise. I did split out the code from `sh:and` and `sh:or` so that it could be used by both classes.